### PR TITLE
doc: Update the roadmap for 1.15

### PR DESCRIPTION
**Pull request by travisn**
For the v1.15 release update the roadmap for new features that are being planned.

**This is still a draft and looking for input from the team**

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Comment by satoru-takeuchi** [comment]
LGTM. It is reasonable to include all listed features in the roadmap. In addition, I verified all links are correct.

**Review by subhamkrai — COMMENTED** [review]
Add cephfs fencing?

File: ROADMAP.md [hunk 1]
```diff
@@ -8,26 +8,16 @@ We hope that the items listed below will inspire further engagement from the com
...
+## Rook Ceph 1.15
 
-The following high level features are targeted for Rook v1.14 (April 2024). For more detailed project tracking see the [v1.14 board](https://github.com/rook/rook/projects/31).
+The following high level features are targeted for Rook v1.15 (July 2024). For more detailed project tracking see the [v1.15 board](https://github.com/rook/rook/projects/32).
 
-* Support for Ceph Squid (v19)
-* Allow setting the application name on a CephBlockPool [#13744](https://github.com/rook/rook/pull/13744)
-* Pool sharing for multiple object stores [#11411](https://github.com/rook/rook/issues/11411)
-* DNS subdomain style access to RGW buckets [#4780](https://github.com/rook/rook/issues/4780)
 * Replace a single OSD when a metadataDevice is configured with multiple OSDs [#13240](https://github.com/rook/rook/issues/13240)
```
Comment (parth-gr) [inline ROADMAP.md:15]: 1) Support logrotate for csi pods https://github.com/rook/rook/issues/12809
2) Support Automatically decide type of MDS pod scaling  https://github.com/rook/rook/issues/14185
3) For external cluster make import and export more easier https://github.com/rook/rook/issues/14077
4) Support key rotation for cephobjectstore user https://github.com/rook/rook/issues/11563

File: ROADMAP.md [hunk 2]
```diff
@@ -8,26 +8,16 @@ We hope that the items listed below will inspire further engagement from the com
...
-* Add topology based provisioning for external clusters [#13821](https://github.com/rook/rook/pull/13821)
+* Multus-enabled clusters deprecate "holder" pods [#13055](https://github.com/rook/rook/issues/13055)
+* Ceph-CSI [v3.12](https://github.com/ceph/ceph-csi/issues?q=is%3Aopen+is%3Aissue+milestone%3Arelease-v3.12.0)
 
 ## Kubectl Plugin
 
-Features are planned in the 1.14 time frame for the [Kubectl Plugin](https://github.com/rook/kubectl-rook-ceph).
+Features are planned for the [Kubectl Plugin](https://github.com/rook/kubectl-rook-ceph), though without a committed timeline.
 * Collect details to help troubleshoot the csi driver [#69](https://github.com/rook/kubectl-rook-ceph/issues/69)
 * Command to flatten an RBD image [#222](https://github.com/rook/kubectl-rook-ceph/issues/222)
```
Comment (BlaineEXE) [inline ROADMAP.md:23]: This is done already, I think?

**Comment by BlaineEXE** [comment]
Roadmap candidate: 
Clean up old logs from de-scheduled daemons pods: #14202

File: ROADMAP.md [hunk 3]
```diff
@@ -8,26 +8,16 @@ We hope that the items listed below will inspire further engagement from the com
...
-* Create a default service account for all Ceph daemons [#13362](https://github.com/rook/rook/pull/13362)
-* Enable the rook orchestrator mgr module by default for improved dashboard integration [#13760](https://github.com/rook/rook/issues/13760)
-* Option to run all components on the host network [#13571](https://github.com/rook/rook/issues/13571)
-* Multus-enabled clusters to begin "holder" pod deprecation [#13055](https://github.com/rook/rook/issues/13055)
-* Separate CSI image repository and tag for all images in the helm chart [#13585](https://github.com/rook/rook/issues/13585)
-* Ceph-CSI [v3.11](https://github.com/ceph/ceph-csi/issues?q=is%3Aopen+is%3Aissue+milestone%3Arelease-v3.11.0)
-* Add build support for Go 1.22 [#13738](https://github.com/rook/rook/pull/13738)
-* Add topology based provisioning for external clusters [#13821](https://github.com/rook/rook/pull/13821)
+* Multus-enabled clusters deprecate "holder" pods [#13055](https://github.com/rook/rook/issues/13055)
+* Ceph-CSI [v3.12](https://github.com/ceph/ceph-csi/issues?q=is%3Aopen+is%3Aissue+milestone%3Arelease-v3.12.0)
```
Comment (Madhu-1) [inline ROADMAP.md:17]: @travisn do we need to add ceph-csi-operator integration tracking as well?

File: ROADMAP.md (hunk 2 above)
Comment (travisn) [reply ROADMAP.md:23]: Yes, it merged since I opened this PR, will remove it

File: ROADMAP.md (hunk 1 above)
Comment (travisn) [reply ROADMAP.md:15]: I'll add the first and fourth. The second and third don't feel defined enough yet to add them to this roadmap doc.

**Review by travisn — COMMENTED** [review]
> Roadmap candidate: Clean up old logs from de-scheduled daemons pods: #14202

@BlaineEXE Unless there is an assignee working on these issues, I'm thinking to not add to the roadmap for now.

File: ROADMAP.md [hunk 4]
```diff
@@ -8,26 +8,16 @@ We hope that the items listed below will inspire further engagement from the com
...
 * Replace a single OSD when a metadataDevice is configured with multiple OSDs [#13240](https://github.com/rook/rook/issues/13240)
-* Create a default service account for all Ceph daemons [#13362](https://github.com/rook/rook/pull/13362)
-* Enable the rook orchestrator mgr module by default for improved dashboard integration [#13760](https://github.com/rook/rook/issues/13760)
-* Option to run all components on the host network [#13571](https://github.com/rook/rook/issues/13571)
-* Multus-enabled clusters to begin "holder" pod deprecation [#13055](https://github.com/rook/rook/issues/13055)
-* Separate CSI image repository and tag for all images in the helm chart [#13585](https://github.com/rook/rook/issues/13585)
-* Ceph-CSI [v3.11](https://github.com/ceph/ceph-csi/issues?q=is%3Aopen+is%3Aissue+milestone%3Arelease-v3.11.0)
-* Add build support for Go 1.22 [#13738](https://github.com/rook/rook/pull/13738)
-* Add topology based provisioning for external clusters [#13821](https://github.com/rook/rook/pull/13821)
+* Multus-enabled clusters deprecate "holder" pods [#13055](https://github.com/rook/rook/issues/13055)
```
Comment (travisn) [inline ROADMAP.md:16]: @BlaineEXE This issue is closed now from 1.14. Is there a different issue to track the 1.15 multus work item to deprecate holder pods?

**Comment by travisn** [comment]
> Add cephfs fencing?

@subhamkrai Is there an issue tracking re-enabling that feature?

File: ROADMAP.md (hunk 4 above)
Comment (BlaineEXE) [reply ROADMAP.md:16]: Holders are deprecated in 1.14. Our plan has been to fully remove holder capabilities from Rook in 1.16, which is presumably also when the new CSI operator will be the default install. But there isn't an item to track that complete removal, no. I'll make one.

Comment (BlaineEXE) [reply ROADMAP.md:16]: We can replace this with

```
* Make go/no-go decision on holder pod removal, and remove holder capabilities if "go" [#14289](https://github.com/rook/rook/issues/14289)
```

**Review by BlaineEXE — APPROVED** [review]

**Pull request merged**